frontend/src/components/Settings/TextInput.tsx

```diff
@@ -0,0 +1,37 @@
+import { component$, useSignal } from '@builder.io/qwik'
+import { getRandomNumString } from '~/utils/getRandomNumString'
+
+type Props = {
+	label: string
+	name?: string
+	description?: string
+	class?: string
+	invalid?: boolean
+	value?: string
+	required?: boolean
+}
+
+export const TextInput = component$<Props>(
+	({ class: className, label, name, description, invalid, value, required }) => {
+		const inputId = useSignal(`${label.replace(/\s+/g, '_')}___${getRandomNumString(15)}`).value
```

Comment:
What's the purpose of the random numbers?

Reply:
if you were to use a `TextInput` with the same label multiple times we wouldn't want the label to point to the incorrect input element

not too likely and we aren't indeed doing that so if I removed this all would work just the same

would you prefer that I remove it?

Reply:
Not a huge deal. I think an id that gets incremented when you add a new input would be a little simpler if that would work.

Reply:
yeah I thought about that one too... but I am not sure if that can cause issues with Qwik... maybe not, I'll give it a try :slightly_smiling_face::+1:

Reply:
![Screenshot 2023-03-01 at 14 16 33](https://user-images.githubusercontent.com/61631103/222166158-096ce7bc-9d81-4b63-a46d-f02861553418.png)

Reply:
PS: I've just replaced `getRandonNumString` with `crypto.randomUUID`

Reply:
sgtm, whatever gets this out the door